nncf/torch/__init__.py

```diff
@@ -10,21 +10,18 @@
  See the License for the specific language governing permissions and
  limitations under the License.
 """
-
+from nncf import nncf_logger
 # pylint: skip-file
 from nncf.version import BKC_TORCH_VERSION
-from nncf.version import BKC_TORCHVISION_VERSION
 
 import torch
 from pkg_resources import parse_version
 torch_version = parse_version(torch.__version__).base_version
 if parse_version(BKC_TORCH_VERSION).base_version != torch_version:
-    import warnings
-    warnings.warn("NNCF provides best results with torch=={bkc}, "
-                   "while current torch version is {curr} - consider switching to torch=={bkc}".format(
-         bkc=BKC_TORCH_VERSION,
-         curr=torch.__version__
-    ))
+    nncf_logger.warn(
+        f"NNCF provides best results with torch=={BKC_TORCH_VERSION}, "
+        f"while current torch version is {torch.__version__} - "
+        f"consider switching to torch=={BKC_TORCH_VERSION}")
```

Comment:
I always find this warning kind of scary, wondering what might go wrong, but usually other PyTorch versions work fine and my understanding is they're expected to work, that the only reason this message is there is because this is the PyTorch version that is validated.

Consider switching is also a pretty big ask, not always possible. Wondering if the message could be something like f"NNCF is validated with torch=={BKC_TORCH_VERSION}, while the current torch version is {torch.__version__}. If you encounter issues, consider switching to torch=={BKC_TORCH_VERSION}"

Reply:
Done